### PR TITLE
Test & Abstract Algorithm for embedding.train.graphwave

### PR DESCRIPTION
**Pull request by paul-tqh-nguyen**
This commit adds a test and abstract algorithm for `embedding.train.graphwave`.

This commit also adds an `as_dense` method to `NumpyVector` that returns a 1-D NumPy array.

### CODE REVIEW

**Comment by paul-tqh-nguyen** [comment]
https://github.com/metagraph-dev/metagraph/pull/136/commits/f3baf8dcffff9da35339e9e4530afe2b2b8d35ad does the following:
* Merges in the latest additions from https://github.com/metagraph-dev/metagraph/pull/134.
* Adds an abstract algorithm for HOPE (using the Katz index).
* Adds an abstract algorithm for GraphSAGE (using the mean aggregator).
* Adds the abstract algorithm for the utility algorithm `util.graph_sage_node_embedding.apply`, which embeds nodes within a given graph using GraphSAGE.
* Adds the property `shape` to `NumpyVector` as well as an `as_dense` method.
* Add partial progress on a HOPE `MultiVerify` test.
* Adds the abstract type `GraphSageNodeEmbedding` and the wrapper `GraphSageNodeEmbeddingWrapper`.

File: metagraph/plugins/numpy/types.py [hunk 1]
```diff
@@ -120,6 +120,18 @@ def __init__(self, data, mask=None):
             if mask.shape != data.shape:
                 raise ValueError("mask must be the same shape as data")
 
+    @property
+    def shape(self):
+        return self.value.shape
+
+    def as_dense(self, fill_value=0, copy=False) -> np.ndarray:
+        vector = self.value
+        if copy:
```
Comment (jim22k) [inline metagraph/plugins/numpy/types.py:129]: This should be `if copy or self.mask is not None`. Even if the user didn't ask for a copy, we never want to mutate the original object.

Comment (jim22k) [reply metagraph/plugins/numpy/types.py:129]: The same applies down to NumpyMatrix.as_dense as well

**Review by jim22k — APPROVED** [review]

File: metagraph/plugins/numpy/types.py (hunk 1 above)
Comment (paul-tqh-nguyen) [reply metagraph/plugins/numpy/types.py]: https://github.com/metagraph-dev/metagraph/pull/136/commits/9541c9057a969101646a6ff571242c4696bd627d implements this suggestion.

**Comment by paul-tqh-nguyen** [comment]
There were some merge issues due to formatting. https://github.com/metagraph-dev/metagraph/pull/136/commits/3635ae9b35e5bddb5124422422f62dc504745328 attempts to fix them.

**Pull request merged**